include/drivers/cavs_ipc.h

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_CAVS_IPC_H
+#define ZEPHYR_INCLUDE_CAVS_IPC_H
+
+#include <cavs-ipc-regs.h>
+#include <kernel.h>
+#include <device.h>
+
+struct cavs_ipc_config {
+	volatile struct cavs_ipc *regs;
+};
+
+/** @brief cAVS IPC Message Handler Callback
+ *
+ * This function, once registered via cavs_ipc_set_message_handler(),
+ * is invoked in interrupt context to service messages sent from the
+ * foreign/connected IPC context.  The message contents of the TDR and
+ * TDD registers are provided in the data/ext_data argument.
+ *
+ * The function should return true if processing of the message is
+ * complete and return notification to the other side (via the TDA
+ * register) is desired immediately.  Returning false means that no
+ * return "DONE" interrupt will occur until cavs_ipc_complete() is
+ * called on this device at some point in the future.
```

Comment:
Hm.... I think I dropped a change here somewhere.  Somewhere I wrote a note about the fact that async DONE interrupts on cAVS 1.5 don't actually work (sorta, it's complicated), but I don't see it in the code as pushed.  Will fix.